app/services/chatbot/chatbot-interfaces.ts

```diff
@@ -0,0 +1,146 @@
+// state
+export interface ChatbotApiServiceState {
```

Comment:
We have a convention of prefixing interfaces with `I`.  So this interface would be called `IChatbotApiServiceState`.  This is something that factions of the TypeScript world inherited from the .NET world I believe.  I'm personally mixed on it, but it's probably best to adhere to the current convention of the codebase until we decide to change it.

Reply:
It just stands for `Interface`.  It seems like the TypeScript community is fairly split about whether it's useful.

Reply:
Typescript documentation doesn't recommend using `I` prefixes for interfaces because the all common interfaces like `HTMLElement`, `Event` don't have this prefix. However many `TS` projects use it, just because of it convenient.